### ISSUE TITLE
Command aliases on Mixin not being applied

### ISSUE THREAD

**Issue by rsenden** [remkop/picocli#1836]
Our discussions in #1665 and #1799 seem to suggest that it should be possible to define command aliases through a Mixin, i.e. something like the following:

```java
@Command(aliases = {"ls"} )
public class ListAliasMixin {}

@Command(name="list")
public class MyListCommand {
    @Mixin private ListAliasMixin aliasMixin; 
}
```

However, for some reason the aliases are not being applied to `MyListCommand`. Any idea why?

**Comment by remkop** [comment]
Yes looks like aliases are not picked up from mixins. I see this in the code for `CommandSpec` (see last line):

```java
            public CommandSpec addMixin(String name, CommandSpec mixin) {
                mixins.put(interpolator.interpolate(name), mixin);

                initName(interpolator.interpolateCommandName(mixin.name()));
                // TODO initAliases(mixin.aliases()); // should we?
```

We can change this. Will you be able to provide a unit test?

**Comment by rsenden** [comment]
I'll look into providing a PR with fix and unit test either myself or by my colleague. 

Interestingly, the `initName` line just above seems to indicate that it should be possible to also initialize the command name through a `Mixin`, however I don't see how this can actually be used as `CommandReflection::subcommandName` explicitly checks whether a subcommand has a `Command` annotation with `name` attribute.

Most of our commands now look as follows, with `MyOutputHelperMixins` providing `Mixin` classes that provide a `CMD_NAME` constant, possibly command aliases (once this issue is fixed), and other generic functionality:
```java
@Command(name = MyOutputHelperMixins.List.CMD_NAME)
public class SomeListCommand extends AbstractOutputCommand implements IBaseHttpRequestSupplier {
    @Getter @Mixin private MyOutputHelperMixins.List outputHelper;
    ...
}
```

If we could specify the command name on the `List` `Mixin`, we could get rid of `@Command(name = MyOutputHelperMixins.List.CMD_NAME)` on every command, either having an empty `@Command` annotation or no `@Command` annotation at all. This would simplify our code, but more importantly, would avoid mismatches like the following (using the command name from `Get`, but using the `List` mixin):

```java
@Command(name = MyOutputHelperMixins.Get.CMD_NAME)
public class SomeListCommand extends AbstractOutputCommand implements IBaseHttpRequestSupplier {
    @Getter @Mixin private MyOutputHelperMixins.List outputHelper;
    ...
}
```

Any suggestions on how to allow the command name to be specified through a `Mixin`? Note that this would fix #1799.
